### PR TITLE
fix prerendered path normalisation

### LINKED ISSUE #5225: Serviceworkers `prerendered` paths incorrect with `trailingSlash: 'always'`
### Describe the bug

When having set `trailingSlash` to `'always'`, the paths for page-endpoints (`__data.json`) and endpoints like `sitemap.xml` in the `prerendered` variable of the `$service-worker` module are normalized to have a trailing slash.

### Reproduction

https://stackblitz.com/edit/sveltejs-kit-template-default-xkpctn
See console output

### Logs

_No response_

### System Info

```shell
Not relevant
```

### Severity

serious, but I can work around it

### Additional Information

I'm not sure, if it would be enough to revert this change: https://github.com/sveltejs/kit/commit/1ea4b06f158a8f52012e4a4cb5152a80b52f943b#diff-693bcce9e16dcf8cee3dc704a2e1c2f2024b1d540e31c3bb050a5112a61ef2d1

### PR DESCRIPTION
**Pull request by Rich-Harris**
fixes #5225

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/core/prerender/prerender.js [hunk 1]
```diff
@@ -285,7 +285,9 @@ export async function prerender({ config, entries, files, log }) {
 							location: resolved
 						});
 
-						prerendered.paths.push(normalize_path(decoded, 'never'));
+						prerendered.paths.push(
+							normalize_path(decoded, file.endsWith('/index.html') ? 'always' : 'never')
```
Comment (dummdidumm) [inline packages/kit/src/core/prerender/prerender.js:289]: Why do we need to handle `/index.html` differently? Do we need to handle `/index.html` differently in all cases, so it should be moved into `normalize_path`?

File: packages/kit/src/vite/build/build_service_worker.js [hunk 2]
```diff
@@ -55,9 +55,7 @@ export async function build_service_worker(
 			];
 
 			export const prerendered = [
-				${prerendered.paths
-					.map((path) => s(normalize_path(path, config.kit.trailingSlash)))
-					.join(',\n\t\t\t\t')}
+				${prerendered.paths.map((path) => s(path)).join(',\n\t\t\t\t')}
```
Comment (dummdidumm) [inline packages/kit/src/vite/build/build_service_worker.js:58]: Just to clarify: We don't normalize here because we already did in `prerender.js`?

**Review by dummdidumm — APPROVED** [review]
Left some questions mainly to clarify some things for myself, looks good otherwise

File: packages/kit/src/core/prerender/prerender.js (hunk 1 above)
Comment (Rich-Harris) [reply packages/kit/src/core/prerender/prerender.js:289]: I started writing an explanation of this but it just made me realise we can simplify the implementation... one sec

File: packages/kit/src/vite/build/build_service_worker.js (hunk 2 above)
Comment (Rich-Harris) [reply packages/kit/src/vite/build/build_service_worker.js:58]: exactly, yep. previously this was normalising pages _and_ endpoints; we only want pages to be normalised, hence moving the logic from here to there

**Pull request merged**